### PR TITLE
Validations Added

### PR DESCRIPTION
**Pull request by ItezazulHassan**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -2,4 +2,12 @@ class Cart < ApplicationRecord
     belongs_to :user
     has_many :line_items
     has_many :items, through: :line_items
+    
+    def sub_total
+        sum=0
+        line_items.each do |item|
+            sum += item.total_price
+        end
+        sum
```
Comment (mshakeelcs) [inline app/models/cart.rb]:
```suggestion
        line_items.sum(&:total_price)
```

File: db/migrate/20210818065350_add_picture_to_items.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddPictureToItems < ActiveRecord::Migration[6.1]
+  def change
+    add_column :items, :picture, :string
```
Comment (mshakeelcs) [inline db/migrate/20210818065350_add_picture_to_items.rb:3]: read about active storage and use it for pics

File: app/models/item.rb [hunk 3]
```diff
@@ -1,6 +1,11 @@
 class Item < ApplicationRecord
-    has_many :item_categories
+    has_many :item_categories, dependent: :destroy
     has_many :categories, through: :item_categories
-    has_many :line_items
+    has_many :line_items, dependent: :destroy
     has_many :carts, through: :line_items
+    mount_uploader :picture, ProductPictureUploader
```
Comment (mshakeelcs) [inline app/models/item.rb:6]: ProductPictureUploader ?

Comment (ItezazulHassan) [reply app/models/item.rb:6]: Its a typo, I will fix it

File: db/migrate/20210818065350_add_picture_to_items.rb (hunk 2 above)
Comment (ItezazulHassan) [reply db/migrate/20210818065350_add_picture_to_items.rb:3]: Alright sir

**Pull request merged**